### PR TITLE
Bitfinex: replace errTypeAssert with common.GetTypeAssertError

### PR DESCRIPTION
**Pull request by Beadko**
# PR Description

Bitfinex: replace errTypeAssert with common.GetTypeAssertError

Fixes # (issue)

## Type of change

Please delete options that are not relevant and add an `x` in `[]` as item is complete.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How has this been tested

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration and
also consider improving test coverage whilst working on a certain feature or package.

- [x] go test ./... -race
- [x] golangci-lint run
- [ ] Test X

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation and regenerated documentation via the documentation tool
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally and on Github Actions/AppVeyor with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: exchanges/bitfinex/bitfinex.go [hunk 1]
```diff
@@ -163,37 +158,32 @@ func symbolMarginInfo(data []interface{}) ([]MarginInfoV2, error) {
...
-		if !ok {
-			return nil, fmt.Errorf("%w for BestAskAmount", errTypeAssert)
+		if tempResp.BestAskAmount, ok = pairMarginInfo[2].(float64); !ok {
+			return nil, common.GetTypeAssertError("float64", pairMarginInfo[2], "MarginInfo.Symbol.BestAskAmount")
 		}
-		tempResp.BestBidAmount, ok = tempFloatData[3].(float64)
-		if !ok {
-			return nil, fmt.Errorf("%w for BestBidAmount", errTypeAssert)
+		if tempResp.BestBidAmount, ok = pairMarginInfo[3].(float64); !ok {
+			return nil, common.GetTypeAssertError("float64", pairMarginInfo[3], "MarginInfo.Symbol.BestBidAmount")
```
Comment (gloriousCode) [inline exchanges/bitfinex/bitfinex.go:186]: Shouldn't this be `MarginInfo.Data...` MarginInfo.Symbol is defined ln166 and also https://docs.bitfinex.com/reference/rest-auth-info-margin

File: exchanges/bitfinex/bitfinex.go [hunk 2]
```diff
@@ -203,32 +193,27 @@ func symbolMarginInfo(data []interface{}) ([]MarginInfoV2, error) {
...
 	}
-	if len(tempData) < 4 {
+	if len(marginInfo) < 4 {
 		return resp, errors.New("invalid data received")
 	}
-	resp.TradableBalance, ok = tempData[0].(float64)
-	if !ok {
-		return resp, fmt.Errorf("%w for TradableBalance", errTypeAssert)
+	if resp.TradableBalance, ok = marginInfo[0].(float64); !ok {
+		return resp, common.GetTypeAssertError("float64", marginInfo[0], "MarginInfo.Symbol.TradableBalance")
```
Comment (gloriousCode) [inline exchanges/bitfinex/bitfinex.go:207]: Same here

File: exchanges/bitfinex/bitfinex.go [hunk 3]
```diff
@@ -655,93 +636,93 @@ func (b *Bitfinex) GetTickerBatch(ctx context.Context) (map[string]Ticker, error
...
 		if !ok {
-			return nil, common.GetTypeAssertError("string", response[x][0], "symbol")
+			return nil, common.GetTypeAssertError("string", response[x][0], "Ticker.Symbol")
 		}
 
 		var t Ticker
 		if len(response[x]) > 11 {
 			if t.FlashReturnRate, ok = response[x][1].(float64); !ok {
-				return nil, common.GetTypeAssertError("float64", response[x][1], "FlashReturnRate")
+				return nil, common.GetTypeAssertError("float64", response[x][1], "Ticker.Symbol.FlashReturnRate")
```
Comment (gloriousCode) [inline exchanges/bitfinex/bitfinex.go:645]: Same here, these aren't part of a `Symbol` struct

**Review by gloriousCode — CHANGES_REQUESTED** [review]
Thanks for doing this @Beadko! Just a couple of really annoying and tiny comments

File: exchanges/bitfinex/bitfinex.go (hunk 2 above)
Comment (Beadko) [reply exchanges/bitfinex/bitfinex.go:211]: Done

File: exchanges/bitfinex/bitfinex.go (hunk 1 above)
Comment (Beadko) [reply exchanges/bitfinex/bitfinex.go]: done

File: exchanges/bitfinex/bitfinex.go (hunk 3 above)
Comment (Beadko) [reply exchanges/bitfinex/bitfinex.go]: done

**Review by shazbert — APPROVED** [review]
LGTM! Thanks Bea.

File: exchanges/bitfinex/bitfinex.go (hunk 2 above)
Comment (gloriousCode) [reply exchanges/bitfinex/bitfinex.go:211]: This one was not actioned

**Review by gloriousCode — CHANGES_REQUESTED** [review]
One more thing then Im happy 🎉

File: exchanges/bitfinex/bitfinex.go (hunk 2 above)
Comment (Beadko) [reply exchanges/bitfinex/bitfinex.go]: Oops, sorry for that. No corrected

**Review by gloriousCode — APPROVED** [review]
tACK! Thanks Beadko

**Review by thrasher- — APPROVED** [review]
All strings match up to their respective vars, thanks @Beadko !

**Pull request merged**